File: frontend/src/types/debt.ts
export enum DebtType {
  Borrowed = 1,
  Lent = 2,
}

export enum DebtStatus {
  Unpaid = 1,
  PartiallyPaid = 2,
  Paid = 3,
}

export interface DebtPayment {
  id: string;
  debtId: string;
  amount: number;
  date: string;
  note?: string;
}

export interface Debt {
  id: string;
  title: string;
  personName: string;
  amount: number;
  remainingAmount: number;
  paidAmount: number;
  type: DebtType;
  typeName: string;
  status: DebtStatus;
  statusName: string;
  dueDate?: string;
  note?: string;
  createdAt: string;
}

export interface CreateDebtDto {
  title: string;
  personName: string;
  amount: number;
  type: DebtType;
  dueDate?: string;
  note?: string;
}

export interface UpdateDebtDto {
  title: string;
  personName: string;
  amount: number;
  type: DebtType;
  dueDate?: string;
  note?: string;
}

export interface CreateDebtPaymentDto {
  amount: number;
  date: string;
  note?: string;
}

export interface DebtSummary {
  totalBorrowed: number;
  totalLent: number;
  totalBorrowedRemaining: number;
  totalLentRemaining: number;
  overdueCount: number;
}

File: ios/ExpenseTrackerApp/Shared/Models/Debt.swift
import Foundation

// MARK: - Enums

enum DebtType: Int, Codable, CaseIterable {
    case borrowed = 1
    case lent = 2

    var displayName: String {
        switch self {
        case .borrowed: return "Đang vay"
        case .lent: return "Đã cho vay"
        }
    }
}

enum DebtStatus: Int, Codable, CaseIterable {
    case unpaid = 1
    case partiallyPaid = 2
    case paid = 3

    var displayName: String {
        switch self {
        case .unpaid: return "Chưa thanh toán"
        case .partiallyPaid: return "Thanh toán một phần"
        case .paid: return "Đã thanh toán"
        }
    }
}

// MARK: - Response Models

struct Debt: Codable, Identifiable {
    let id: String
    let title: String
    let personName: String
    let originalAmount: Double
    let remainingAmount: Double
    let paidAmount: Double
    let type: DebtType
    let typeName: String
    let status: DebtStatus
    let statusName: String
    let dueDate: String?
    let note: String?
    let createdAt: String

    var isOverdue: Bool {
        guard let due = dueDate?.asDate, status != .paid else { return false }
        return due < Date()
    }
}

struct DebtPayment: Codable, Identifiable {
    let id: String
    let debtId: String
    let amount: Double
    let paidDate: String
    let note: String?
    let createdAt: String?
}

struct DebtSummary: Codable {
    let totalBorrowed: Double
    let totalLent: Double
    let totalBorrowedRemaining: Double
    let totalLentRemaining: Double
    let overdueCount: Int
}

// MARK: - Request Models

struct CreateDebtRequest: Codable {
    let title: String
    let personName: String
    let originalAmount: Double
    let type: Int
    let dueDate: String?
    let note: String?
}

struct UpdateDebtRequest: Codable {
    let title: String
    let personName: String
    let dueDate: String?
    let note: String?
}

struct CreateDebtPaymentRequest: Codable {
    let amount: Double
    let paidDate: String
    let note: String?
}
